src/flow_graph.py

```diff
@@ -1095,16 +1277,216 @@ def reset_block_info(self) -> None:
         for node in self.nodes:
             node.block.block_info = None
 
+    def add_dependency(self, *, dst: Reference, loc: Location, src: Reference) -> None:
+        """Update instr_inputs/instr_uses that `dst` depends on `loc` from `src`"""
```

Comment:
I just realized I've been misreading the direction of this in all the callers, reading "dst: a, src: b" as "there is a dependency arrow from b to a" (shows how easily my eyes glance over when reading graph manipulation code that I noticed this only now, hah). Can we find better names than src/dst?